mysql-test/t/audit_log_default_db.test

```diff
@@ -0,0 +1,74 @@
+# test correctness of default_db field
+
+--source include/not_embedded.inc
+
+let $MYSQLD_DATADIR= `select @@datadir`;
+let $log_file=$MYSQLD_DATADIR/test_audit.log;
+
+CREATE USER 'user1'@'%' IDENTIFIED BY '111';
+CREATE USER 'user2'@'%' IDENTIFIED BY '111';
+
+CREATE DATABASE db1;
+CREATE DATABASE db2;
+
+CREATE TABLE db1.t (a VARCHAR(4));
+CREATE TABLE db2.t (a VARCHAR(4));
+INSERT INTO db1.t VALUES ('db1');
+INSERT INTO db2.t VALUES ('db2');
+
+GRANT ALL PRIVILEGES ON db1.* TO 'user1'@'%';
+
+GRANT ALL PRIVILEGES ON db2.* TO 'user1'@'%';
+GRANT ALL PRIVILEGES ON db2.* TO 'user2'@'%';
+
+# truncate audit log
+--remove_file $log_file
+SET GLOBAL audit_log_flush=ON;
+
+connect (test,localhost,user1,111,db2,);
+connection test;
+SELECT * FROM t;
+use 	`db1`;
+--error ER_DBACCESS_DENIED_ERROR
+change_user user2,111,db1;
+disconnect test;
+--source include/wait_until_disconnected.inc
+--replace_result $MASTER_MYPORT MASTER_MYPORT $MASTER_MYSOCK MASTER_MYSOCK
+--error ER_DBACCESS_DENIED_ERROR
+connect (test,localhost,user2,111,db1,);
+--source include/wait_until_disconnected.inc
+--replace_result $MASTER_MYPORT MASTER_MYPORT $MASTER_MYSOCK MASTER_MYSOCK
+--error ER_ACCESS_DENIED_ERROR
+connect (test,localhost,user2,112,db2,);
+--source include/wait_until_disconnected.inc
+--replace_result $MASTER_MYPORT MASTER_MYPORT $MASTER_MYSOCK MASTER_MYSOCK
+--error ER_ACCESS_DENIED_ERROR
+connect (test,localhost,user3,111,db2,);
+--source include/wait_until_disconnected.inc
+connect (test,localhost,user2,111,db2,);
+connection test;
+SELECT * FROM t;
+--error ER_DBACCESS_DENIED_ERROR
+use    	`db1`;
+SELECT * FROM t;
+change_user user1,111,db1;
+SELECT * FROM t;
+use `db2`;
+SELECT * FROM t;
+disconnect test;
+--source include/wait_until_disconnected.inc
+
+connection default;
+
+--exec $MYSQL --user=user1 --password=111 test -e "use db1; SELECT * FROM t;"
+# is there better way to check that server reset connection initiated by mysql cli?
```

Comment:
Wrapping exec with include/count_connections and include/wait_until_connection_count should do this